services/API-service/src/api/event/event.service.ts

```diff
@@ -69,38 +71,67 @@ export class EventService {
     disasterType: DisasterType,
   ): Promise<EventSummaryCountry[]> {
     const recentDate = await this.getRecentDate(countryCodeISO3, disasterType);
-    const eventSummary = await this.eventPlaceCodeRepo
-      .createQueryBuilder('event')
-      .select([
-        'area."countryCodeISO3"',
-        'event."eventName"',
-        'event."triggerValue"',
-      ])
-      .leftJoin('event.adminArea', 'area')
-      .groupBy('area."countryCodeISO3"')
-      .addGroupBy('event."eventName"')
-      .addGroupBy('event."triggerValue"')
-      .addSelect([
-        'to_char(MIN("startDate") , \'yyyy-mm-dd\') AS "startDate"',
-        'to_char(MAX("endDate") , \'yyyy-mm-dd\') AS "endDate"',
-        'MAX(event."thresholdReached"::int)::boolean AS "thresholdReached"',
-      ])
-      .where({
-        closed: false,
-        endDate: MoreThanOrEqual(recentDate.date),
-        disasterType: disasterType,
-      })
-      .andWhere('area."countryCodeISO3" = :countryCodeISO3', {
-        countryCodeISO3: countryCodeISO3,
-      })
-      .getRawMany();
+    const eventSummaryQueryBuilder = this.createEventSummaryQueryBuilder(
+      countryCodeISO3,
+    ).andWhere({
+      closed: false,
+      endDate: MoreThanOrEqual(recentDate.date),
+      disasterType: disasterType,
+    });
+    const rawEventSummary = await eventSummaryQueryBuilder.getRawMany();
 
-    const disasterSettings = await this.getCountryDisasterSettings(
+    const eventSummary = await this.populateEventsDetails(
+      rawEventSummary,
+      countryCodeISO3,
+      disasterType,
+    );
+    return eventSummary;
+  }
+
+  public async getEventsSummaryTriggerFinishedMail(
+    countryCodeISO3: string,
+    disasterType: DisasterType,
+  ): Promise<EventSummaryCountry[]> {
+    const countryAdminAreaIds = await this.getCountryAdminAreaIds(
+      countryCodeISO3,
+    );
+
+    // I spend quite some time on trying to figure out what is the right query to get the event finished summary for the trigger closed email
+    // I came up with the following query but I am not sure if it is correct and how to test it properly
+    const sevenDaysAgo = new Date(Date.now() - 7 * 24 * 60 * 60 * 1000);
+    const sixDaysAgo = new Date(Date.now() - 6 * 24 * 60 * 60 * 1000);
+    const whereFilters = {
+      endDate: Between(sevenDaysAgo, sixDaysAgo),
+      adminArea: In(countryAdminAreaIds),
+      disasterType: disasterType,
+      closed: true,
+    };
```

Comment:
> the logic here is different from [getFinishedEvent](https://github.com/rodekruis/IBF-system/blob/c879e0673e78859f5eab05510c0ab6940eb5de38/services/API-service/src/api/notification/notification.service.ts#L75-L91). I suggest to confirm this new behaviour with the team before merge. Bcoz we've discussed this I agree your interpretation, the only thing that strongly recommend is to keep the check binary instead of a range. i.e. use LessThan instead of Between.

I did not get to checking this with the rest of the team unfortunately